src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -364,11 +357,11 @@ class TaskExecutor {
   std::shared_ptr<metrics::Meter> throughput_meter_;
   boost::shared_mutex model_queues_mutex_;
   std::unordered_map<const VersionedModelId, std::shared_ptr<ModelQueue>,
-                     decltype(&versioned_model_hash)>
+                     std::hash<clipper::VersionedModelId>>
```

Comment:
I think the reason you need to provide the hash function explicitly here is because the keys are marked const. Does this work if you remove const?

```cpp
std::unordered_map<VersionedModelId, std::shared_ptr<ModelQueue>> model_queues_;
```